.github/workflows/github.yml

```diff
@@ -229,14 +283,20 @@ jobs:
 
     - name: Configure
       run: |
-        if [[ (${{matrix.preset}} == android-conan-ninja-release) && (${{github.ref}} != 'refs/heads/master') ]]
-        then
-            cmake -DENABLE_CCACHE:BOOL=ON -DANDROID_GRADLE_PROPERTIES="applicationIdSuffix=.daily;signingConfig=dailySigning;applicationLabel=VCMI daily;applicationVariant=daily" --preset ${{ matrix.preset }}
-        elif ${{startsWith(matrix.platform, 'msvc') }}
-        then
-            cmake --preset ${{ matrix.preset }}
+        if [[ ("${{ matrix.preset }}" == "android-conan-ninja-release") && ("${{ github.ref }}" != 'refs/heads/master') ]]; then
+          cmake -DENABLE_CCACHE:BOOL=ON \
+                -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+                -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+                -DANDROID_GRADLE_PROPERTIES="applicationIdSuffix=.daily;signingConfig=dailySigning;applicationLabel=VCMI daily;applicationVariant=daily" \
+                --preset ${{ matrix.preset }}
+        elif ${{ startsWith(matrix.platform, 'msvc') }}; then
+            CL="$VCToolsInstallDir/bin/${{ matrix.cl }}"
+            cmake \
+              -D CMAKE_C_COMPILER:FILEPATH="$CL" \
+              -D CMAKE_CXX_COMPILER:FILEPATH="$CL" \
```

Comment:
why is this required? Can't CMake find proper compiler based on the architecture variant requested?

Reply:
Exactly, without this you can't build it. Because mismatches x64 vs x86 etc. have not found better way to compile.